src/header/common/link.rs

```diff
@@ -0,0 +1,1141 @@
+
+        let expected_link = Link::new(vec![link_value]);
+
+        let link = Header::parse_header(&vec![link_header.to_vec()].into());
+        assert_eq!(link.ok(), Some(expected_link));
+    }
+
+    #[test]
+    fn test_link_multiple_link_headers() {
+        let first_link = LinkValue::new("/TheBook/chapter2")
+            .push_rel(RelationType::Previous)
+            .set_title_star("UTF-8'de'letztes%20Kapitel");
+
+        let second_link = LinkValue::new("/TheBook/chapter4")
+            .push_rel(RelationType::Next)
+            .set_title_star("UTF-8'de'n%c3%a4chstes%20Kapitel");
+
+        let third_link = LinkValue::new("http://example.com/TheBook/chapter2")
+            .push_rel(RelationType::Previous)
+            .push_rev(RelationType::Next)
+            .set_title("previous chapter");
+
+        let expected_link = Link::new(vec![first_link, second_link, third_link]);
+
+        let raw = MemBuf::from(b"GET /super_short_uri/and_whatever HTTP/1.1\r\nHost: \
+                                  hyper.rs\r\nAccept: a lot of things\r\nAccept-Charset: \
+                                  utf8\r\nAccept-Encoding: *\r\nLink: </TheBook/chapter2>; \
+                                  rel=\"previous\"; title*=UTF-8'de'letztes%20Kapitel, \
+                                  </TheBook/chapter4>; rel=\"next\"; title*=\
+                                  UTF-8'de'n%c3%a4chstes%20Kapitel\r\n\
+                                  Access-Control-Allow-Credentials: None\r\nLink: \
+                                  <http://example.com/TheBook/chapter2>; \
+                                  rel=\"previous\"; rev=next; title=\"previous chapter\"\
+                                  \r\n\r\n".to_vec());
+
+        let (mut res, _) = ServerTransaction::parse(&raw).unwrap().unwrap();
+
+        let link = res.headers.remove::<Link>().unwrap();
+
+        assert_eq!(link, expected_link);
+    }
+
+    #[test]
+    fn test_link_display() {
+        let link_value = LinkValue::new("http://example.com/TheBook/chapter2")
+            .push_rel(RelationType::Previous)
+            .set_anchor("/anchor/example/")
+            .push_rev(RelationType::Next)
+            .push_href_lang(langtag!(de))
+            .push_media_desc(MediaDesc::Screen)
+            .set_title("previous chapter")
+            .set_title_star("title* unparsed")
+            .set_media_type(Mime(Text, Plain, vec![]))
+            .set_link_extension("link-extension unparsed");
+
+        let link = Link::new(vec![link_value]);
+
+        let mut link_header = String::new();
+        write!(&mut link_header, "{}", link).unwrap();
+
+        let expected_link_header = "<http://example.com/TheBook/chapter2>; \
+            rel=\"previous\"; anchor=\"/anchor/example/\"; \
+            rev=\"next\"; hreflang=de; media=\"screen\"; \
+            title=\"previous chapter\"; title*=title* unparsed; \
+            type=\"text/plain\"; link-extension=link-extension unparsed";
```

Comment:
If that is indeed correct, `LinkValue`'s `link_extension` field should probably be named `link_extensions` and return a `Option<Vec<LinkExtension>>` (or just a `Vec<LinkExtension>`) where `LinkExtension` is something like

```rust
struct LinkExtension {
  name: String,
  value: Option<String>
}
```